### PR TITLE
Require JDK 17 for build

### PR DESCRIPTION
**Pull request by slawekjaranowski**
- reuse javaVersion property from parent
- property maven.compiler.release is set by parent for JDK 11+
- reuse checking of bytecode version from parent

### CODE REVIEW

**Comment by slawekjaranowski** [comment]
Required for #326 - newer version of eclipse compiler

**Review by slachiewicz — APPROVED** [review]

**Comment by slawekjaranowski** [comment]
We need a newer Maven 3.9.6+ due to old sisu without support for JDK 11+

**Comment by slachiewicz** [comment]
To build, test the eclipse compiler component. But later all other compilers can be used normally. 
Maybe just add note to release notes?

**Comment by slawekjaranowski** [comment]
> To build, test the eclipse compiler component. But later all other compilers can be used normally. Maybe just add note to release notes?

Good idea 👍

**Review by slachiewicz — APPROVED** [review]

File: plexus-compilers/plexus-compiler-eclipse/pom.xml [hunk 1]
```diff
@@ -14,7 +14,7 @@
   <description>Eclipse Compiler support for Plexus Compiler component.</description>
 
   <properties>
-    <maven.compiler.release>11</maven.compiler.release>
+    <javaVersion>17</javaVersion>
```
Comment (pzygielo) [inline plexus-compilers/plexus-compiler-eclipse/pom.xml:17]: Not yet sure why, but this breaks using eclipse compiler. With 17 here I'm getting 
```
No such compiler 'eclipse'.
```
Once set to 11 - ecj works fine again.
Please restore.

Comment (slachiewicz) [reply plexus-compilers/plexus-compiler-eclipse/pom.xml:17]: You need a newer Maven (Sosu) see release notes

Comment (pzygielo) [reply plexus-compilers/plexus-compiler-eclipse/pom.xml:17]: @slachiewicz Thanks - I've completely missed that in RN.

**Pull request merged**